flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -379,6 +391,40 @@ public class NettyShuffleEnvironmentOptions {
                                     + "tasks have occupied all the buffers and the downstream tasks are waiting for the exclusive buffers. The timeout breaks"
                                     + "the tie by failing the request of exclusive buffers and ask users to increase the number of total buffers.");
 
+    /** The option to enable the new mode of hybrid shuffle. */
+    @Documentation.Section(Documentation.Sections.ALL_TASK_MANAGER_NETWORK)
+    @Experimental
+    public static final ConfigOption<Boolean> NETWORK_HYBRID_SHUFFLE_ENABLE_NEW_MODE =
+            ConfigOptions.key(HYBRID_SHUFFLE_NEW_MODE_OPTION_NAME)
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription(
+                            "The option is used to enable the new mode of hybrid shuffle, which has resolved existing issues in the legacy mode. First, the new mode "
+                                    + "uses less required network memory. For large parallelism jobs, less network memory is required. Second, the new mode "
+                                    + "can store shuffle data in remote storage when the disk space is not enough, which could avoid insufficient disk space "
+                                    + "errors and is only supported when "
+                                    + HYBRID_SHUFFLE_REMOTE_STORAGE_BASE_PATH_OPTION_NAME
+                                    + " is configured. "
+                                    + "The option is set true by default, but the new mode can fallback to legacy mode by setting this option false. "
+                                    + "In the future, the new mode will become the default implementation of hybrid shuffle and this option will be removed.");
```

Comment:
1. "First, the new mode uses less required network memory. For large parallelism jobs, less network memory is required." This sounds duplicated.
2. "but the new mode can fallback to legacy mode by setting this option false." Why would user fallback to the legacy mode?
3. "In the future, the new mode will become the default implementation of hybrid shuffle and this option will be removed." The new mode is already the default implementation.